### DISCUSSION TITLE
How to stop prisma plugin auto-adding relation

### DISCUSSION THREAD

**Discussion by pitops** [prisma/prisma#20994] [Q&A]
### Question

```prisma
model Size {
  id        String   @id @default(uuid())
  name      String
  value     String
  storeId   String
  store     Store    @relation("StoreToSize", fields: [storeId], references: [id])
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  Store     Store    @relation(fields: [storeId], references: [id]) << adds it again

  @@index([storeId])
}
```

Prisma adds the relation again. I have auto-save on, and i am not going to disable. This is a bug. 

### Expected behavior (optional)

```prisma
model Size {
  id        String   @id @default(uuid())
  name      String
  value     String
  storeId   String
  store     Store    @relation("StoreToSize", fields: [storeId], references: [id])
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  @@index([storeId])
}
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @pitops 👋 

I am unable to reproduce this behavior. Please let us know what version of Prisma you are using.

**Reply by ludralph** [reply]
Also, Prisma always requires back relations to be present. Can you please share your Store model?

**Reply by MaQianheng** [reply]
Hi @ludralph 
I faced same problem. As showed photo below, domesticLogisticCert is missing an opposite relation field which I knwo, I will add it later, manually.
<img width="1137" height="897" alt="Snipaste_2025-09-03_09-29-15" src="https://github.com/user-attachments/assets/b093df7f-ab14-44e6-8cbf-597fe516f542" />
If I save this file, it will auto add missing relation as shown below. But this is fine, it is wrong, VS Code will tell me the error and I can see it and fix it. The problem is that sometimes there is no syntax error, so I will forget it. The auto fixed code usually is wrong and not what I want. How to disable it?
<img width="1153" height="922" alt="Snipaste_2025-09-03_09-29-51" src="https://github.com/user-attachments/assets/48a269bf-64c7-4259-ad09-a7fba57bd929" />
Here is my Prisma version.
<img width="498" height="312" alt="Snipaste_2025-09-03_09-33-15" src="https://github.com/user-attachments/assets/c9af803c-4a78-4a01-bb92-693297740c04" />

**Comment by masterbater** [comment]
You might have defined two relationship in your Store table

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
